### PR TITLE
Add BeeFore build trigger.

### PR DESCRIPTION
**Pull request by freakboy3742**

### CODE REVIEW

File: voc/python/modules.py [hunk 1]
```diff
@@ -14,7 +14,8 @@
 
 class ModuleBlock(Block):
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs
+            )
```
Comment (freakboy3742) [inline voc/python/modules.py]: Test comment.

File: voc/python/modules.py [hunk 2]
```diff
@@ -14,7 +14,8 @@
```
Comment (brutusthebee) [inline voc/python/modules.py]: response

File: voc/python/modules.py [hunk 3]
```diff
@@ -14,7 +14,8 @@
 
 class ModuleBlock(Block):
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
```
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12, found [E124: closing bracket does not match visual indentation](http://.../E124)

Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py [hunk 4]
```diff
@@ -13,16 +13,20 @@
 
 
 class START_LOOP:
-    """A pseudo instruction that can be used to mark the
```
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/modules.py (hunk 3 above)
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py (hunk 4 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/modules.py (hunk 3 above)
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py (hunk 4 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/modules.py [hunk 5]
```diff
@@ -14,7 +14,8 @@
...
+        super().__init__(*args, **kwargs
```
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py [hunk 6]
```diff
@@ -13,16 +13,20 @@
...
+    """
+
+    A pseudo instruction that can be used to mark the
     start of a loop (either a FOR or a WHILE)
     """
     def __init__(self):
         # The opcode that is the first in the formal LOOP block
         self.start_op = None
 
+
```
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/opcodes.py [hunk 7]
```diff
@@ -13,16 +13,20 @@
...
         # The last operation in the LOOP block
         self.end_op = None
 
+
```
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/modules.py (hunk 5 above)
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py (hunk 6 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/opcodes.py (hunk 7 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/modules.py (hunk 5 above)
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py (hunk 6 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/opcodes.py (hunk 7 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/modules.py (hunk 5 above)
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py (hunk 6 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/opcodes.py (hunk 7 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/sample.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+var module = angular.module('uploader', []);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 14: [(no-undef) 'angular' is not defined.](http://.../no-undef)

File: voc/sample.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * XMLHttpRequest wrapper that supports file upload progress since $http doesn't
+ *
+ * Usage similar to $http, returns a promise from the send method
+ */
+
+module.service('uploader', ['$q', function($q) {
+  function readyStateChange(deferred, xhr) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 3: [(indent) Expected indentation of 4 space characters but found 2.](http://.../indent)

File: voc/sample.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    if(xhr.readyState == 4) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 5: [(indent) Expected indentation of 6 space characters but found 4.](http://.../indent)

File: voc/sample.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+      if(xhr.status == 200) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        deferred.resolve(JSON.parse(xhr.responseText));
```
Comment (brutusthebee) [inline voc/sample.js]: At column 9: [(indent) Expected indentation of 10 space characters but found 8.](http://.../indent)

File: voc/sample.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+      }
+      else {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+        deferred.reject('HTTP status ' + xhr.status);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 9: [(indent) Expected indentation of 10 space characters but found 8.](http://.../indent)

File: voc/sample.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+      }
+    }
+  }
+
+  function onProgress(deferred, xhr, ev) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 3: [(indent) Expected indentation of 4 space characters but found 2.](http://.../indent)

File: voc/sample.js [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    if(ev.lengthComputable) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 5: [(indent) Expected indentation of 6 space characters but found 4.](http://.../indent)

File: voc/sample.js [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+      deferred.notify({ loaded: ev.loaded, total: ev.total });
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+    }
+  }
+
+  return {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 3: [(indent) Expected indentation of 4 space characters but found 2.](http://.../indent)

File: voc/sample.js [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+    send: function(url, data) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 5: [(indent) Expected indentation of 6 space characters but found 4.](http://.../indent)

File: voc/sample.js [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+      var fd = new FormData();
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+      for(var k in data) {
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+        fd.append(k, data[k]);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 9: [(indent) Expected indentation of 10 space characters but found 8.](http://.../indent)

File: voc/sample.js [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+      }
+
+      var d = $q.defer();
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+
+      var xhr = new XMLHttpRequest();
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+
+      xhr.open('POST', url, true);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+      xhr.onreadystatechange = readyStateChange.bind({}, d, xhr);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+      xhr.upload.onprogress = onProgress.bind({}, d, xhr);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+      xhr.send(fd);
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/sample.js [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+
+      return d.promise;
```
Comment (brutusthebee) [inline voc/sample.js]: At column 7: [(indent) Expected indentation of 8 space characters but found 6.](http://.../indent)

File: voc/python/modules.py (hunk 5 above)
Comment (brutusthebee) [inline voc/python/modules.py]: At column 12: [(E124) closing bracket does not match visual indentation](http://.../E124)

File: voc/python/opcodes.py (hunk 6 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

File: voc/python/opcodes.py (hunk 7 above)
Comment (brutusthebee) [inline voc/python/opcodes.py]: At column 8: [(E303) too many blank lines (2)](http://.../E303)

**Comment by MarkusH** [comment]
I had a discussion with @IanLee1521 at PyCon over dinner about adopting to a common configuration system for Python's linting tools (pycodestyle, flake8, pylint, isort, ...). We two came to the conclusion that a single file used by all projects (e.g. `setup.cfg` as already supported by most of them) might be a good idea. I adopted my packages to follow that pattern. See https://github.com/MarkusH/django-dynamic-forms/blob/master/setup.cfg for an example.

**Comment by freakboy3742** [comment]
Having a proliferation of config files in different formats doesn't help anyone; adopting a common format sounds like a great idea.

**Comment by IanLee1521** [comment]
In addition, I think a common library to handle the parsing and locating of such files would be great too.

For instance, a common way across packages to handle discovery and merging of: system (/etc), user (~/), local (./) and command line configuration files and options would also be great. Specifically, which take precedence and where they should be located (and how to name them).

**Comment by goanpeca** [comment]
It seems this PR is no longer relevant @freakboy3742, should we close it?

**Comment by freakboy3742** [comment]
Yes, this has now been implemented.

**Pull request closed without merging**